gui/src/main/java/kcl/teamIndexZero/traffic/gui/SimulationImageProducer.java

```diff
@@ -38,24 +37,30 @@ public SimulationImageProducer(SimulationMap map, BiConsumer<BufferedImage, Simu
 
     }
 
+    private double alpha = 0;
+
     /**
      * {@inheritDoc}
      */
     @Override
     public void tick(SimulationTick tick) {
+        alpha += 0.1;
         graphics.setBackground(Color.WHITE);
         graphics.clearRect(0, 0, image.getWidth(), image.getHeight());
-
-        map.getObjectsOnMap().forEach(object -> {
-            graphics.setColor(object.getColor());
-            MapPosition pos = object.getPosition();
-            graphics.fillRect(
-                    pos.x,
-                    pos.y,
-                    pos.width,
-                    pos.height
-            );
-        });
+        new GUI_Primitives().drawSmallCar(70,70,alpha,graphics);
+//        new GUI_Primitives().drawSmallCar(70,70,alpha,graphics);
+        new GUI_Primitives().drawSingleRoad(10, 10, 200, 10, 10, 50, 200, 50, graphics);
```

Comment:
Well then what do you suggest? I don't know of a way in which I can help here, other than removing the calling for my functions, if that helps. Sorry, not too great with merging and it's after effects.                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the EE network.                                                                                                                                                                                                                From: Alexander CherednichenkoSent: Sunday 20 March 2016 23:19To: teamIndexZero/index_zero_trafficsystemReply To: teamIndexZero/index_zero_trafficsystemCc: kumar awijeetSubject: Re: [index_zero_trafficsystem] Gui graphics kumar awijeet (#9)In gui/src/main/java/kcl/teamIndexZero/traffic/gui/SimulationImageProducer.java:
>          graphics.setBackground(Color.WHITE);
>          graphics.clearRect(0, 0, image.getWidth(), image.getHeight());
> -
> -        map.getObjectsOnMap().forEach(object -> {
> -            graphics.setColor(object.getColor());
> -            MapPosition pos = object.getPosition();
> -            graphics.fillRect(
> -                    pos.x,
> -                    pos.y,
> -                    pos.width,
> -                    pos.height
> -            );
> -        });
> +        new GUI_Primitives().drawSmallCar(70,70,alpha,graphics);
> +//        new GUI_Primitives().drawSmallCar(70,70,alpha,graphics);
> +        new GUI_Primitives().drawSingleRoad(10, 10, 200, 10, 10, 50, 200, 50, graphics);

So, as soon as we merge, the whole thing which is there will disappear, and instead we will have the car. Just a note, we anyway need to merge master -> GUI branch first, and then see how to use it. 

—You are receiving this because you authored the thread.Reply to this email directly or view it on GitHub